regex/tests/Test.lean

```diff
@@ -153,4 +153,907 @@ def word_class := Regex.parse! r##"\w+"##
+#guard non_word_boundary_18.bt.captureAll "x" = #[]
+
+def non_word_boundary_19 := Regex.parse! r##"^\B$"##
+
+-- name = "nb33"
+#guard non_word_boundary_19.captureAll "" = #[
+  ⟨"", #[.some ⟨0⟩, .some ⟨0⟩]⟩
+]
+#guard non_word_boundary_19.bt.captureAll "" = #[
+  ⟨"", #[.some ⟨0⟩, .some ⟨0⟩]⟩
+]
+
+-- name = "nb34"
+#guard non_word_boundary_19.captureAll "x" = #[]
+#guard non_word_boundary_19.bt.captureAll "x" = #[]
+
+def non_word_boundary_20 := Regex.parse! r##"^\B.$"##
+
+-- name = "nb35"
+#guard non_word_boundary_20.captureAll "x" = #[]
+#guard non_word_boundary_20.bt.captureAll "x" = #[]
+
+def non_word_boundary_21 := Regex.parse! r##"^\B.\B$"##
+
+-- name = "nb36"
+#guard non_word_boundary_21.captureAll "x" = #[]
+#guard non_word_boundary_21.bt.captureAll "x" = #[]
+
+def non_word_boundary_22 := Regex.parse! r##"^^^^^\B$$$$$"##
+
+-- name = "nb37"
+#guard non_word_boundary_22.captureAll "" = #[
+  ⟨"", #[.some ⟨0⟩, .some ⟨0⟩]⟩
+]
+#guard non_word_boundary_22.bt.captureAll "" = #[
+  ⟨"", #[.some ⟨0⟩, .some ⟨0⟩]⟩
+]
+
+def non_word_boundary_23 := Regex.parse! r##"^^^^^\B.$$$$$"##
+
+-- name = "nb38"
+#guard non_word_boundary_23.captureAll "x" = #[]
+#guard non_word_boundary_23.bt.captureAll "x" = #[]
+
+def non_word_boundary_24 := Regex.parse! r##"^^^^^\B$$$$$"##
+
+-- name = "nb39"
+#guard non_word_boundary_24.captureAll "x" = #[]
+#guard non_word_boundary_24.bt.captureAll "x" = #[]
+
+-- unicode tests
+-- unicode1* and unicode2* work for both Unicode and ASCII because all matches
+-- are reported as byte offsets, and « and » do not correspond to word
+-- boundaries at either the character or byte level.
+
+--
+-- Unicode word boundary tests
+--
+def unicode_word_boundary_01 := Regex.parse! r##"\bx\b"##
+
+-- name = "unicode1"
+#guard unicode_word_boundary_01.captureAll "«x" = #[
+  ⟨"«x", #[.some ⟨2⟩, .some ⟨3⟩]⟩
+]
+#guard unicode_word_boundary_01.bt.captureAll "«x" = #[
+  ⟨"«x", #[.some ⟨2⟩, .some ⟨3⟩]⟩
+]
+
+-- name = "unicode1-only-ascii"
+#guard unicode_word_boundary_01.captureAll "«x" = #[
+  ⟨"«x", #[.some ⟨2⟩, .some ⟨3⟩]⟩
+]
+#guard unicode_word_boundary_01.bt.captureAll "«x" = #[
+  ⟨"«x", #[.some ⟨2⟩, .some ⟨3⟩]⟩
+]
+
+-- name = "unicode2"
+#guard unicode_word_boundary_01.captureAll "x»" = #[
+  ⟨"x»", #[.some ⟨0⟩, .some ⟨1⟩]⟩
+]
+#guard unicode_word_boundary_01.bt.captureAll "x»" = #[
+  ⟨"x»", #[.some ⟨0⟩, .some ⟨1⟩]⟩
+]
+
+-- name = "unicode2-only-ascii"
+#guard unicode_word_boundary_01.captureAll "x»" = #[
+  ⟨"x»", #[.some ⟨0⟩, .some ⟨1⟩]⟩
+]
+#guard unicode_word_boundary_01.bt.captureAll "x»" = #[
+  ⟨"x»", #[.some ⟨0⟩, .some ⟨1⟩]⟩
+]
+
+-- ASCII word boundaries are completely oblivious to Unicode characters, so
+-- even though β is a character, an ASCII \b treats it as a word boundary
+-- when it is adjacent to another ASCII character. (The ASCII \b only looks
+-- at the leading byte of β.) For Unicode \b, the tests are precisely inverted.
+
+-- FIXME: This test is not working as expected.
```

Comment:
```suggestion
-- FIXME: This test is not working as expected because we only check for ASCII alphanumeric characters.
```

The reason the test fails is that [`Char.isAlphanum`](https://lean-lang.org/doc/reference/latest//Basic-Types/Characters/#Char___isAlphanum) only look for the ASCII alphanumeric characters while the test expects characters like `á` and `β` to be (Unicode) alphanumeric. Since I plan to introduce such Unicode character handling at a later point, can you rewrite the comment a bit?

Reply:
done.